metricflow-semantics/metricflow_semantics/model/semantics/linkable_element.py

```diff
@@ -222,7 +222,10 @@ def reference(self) -> GroupByMetricReference:  # noqa: D102
 
     @property
     def join_by_semantic_model(self) -> Optional[SemanticModelReference]:  # noqa: D102
-        return self.join_path.last_semantic_model_reference
+        if len(self.join_path.semantic_model_join_path.path_elements) == 0:
```

Comment:
Should this return `self.join_path.semantic_model_join_path.left_semantic_model_reference` instead of `None`? Depends how it's used so I'll leave that up to you.

Reply:
Actually, this is no longer used, so I've removed it.